Various Azure IoT Hub doc improvements (#6238)
nrfconnect/sdk-nrf: doc/nrf/libraries/networking/azure_iot_hub.rst

See `Creating an Azure IoT Hub instance using the Azure portal`_ for more information.

.. note::
   If you do not use DPS to provision devices to your IoT Hub, make sure that you select ``X.509 CA Signed`` as the *Authentication type* while `Registering the device with Azure IoT Hub`_.

The connection to Azure IoT Hub with MQTT is secured using TLS.
For testing purposes, see `Creating Azure IoT Hub certificates`_ for the steps to create certificates and a private key for the leaf device, and to register the generated test root certificate to be used with an IoT hub.

The Azure IoT Hub library requires the provisioning of the following certificates and a private key for a successful TLS connection:

1. `Baltimore CyberTrust Root certificate`_ - Server certificate, used to verify the server's certificate while connecting.
#. Public device certificate - generated by the procedures described in `Creating Azure IoT Hub certificates`_ , used by Azure IoT Hub to authenticate the device.
#. Private key of the device.

.. note::

>    The location and name of the public device certificate and private key files that are generated will vary with what method you use for the credential generation.
>    For PowerShell scripts, the device certificate is called ``mydevice-public.pem`` and the private key ``mydevice-private.pem`` and they are located together with the other generated files in the working directory.

Reviewer: ```suggestion
   The location and name of the generated public device certificate and private key files vary depending on the method you use for the credential generation.
   For PowerShell scripts, the device certificate is called :file:`mydevice-public.pem` and the private key :file:`mydevice-private.pem`.
   These files are located  in the working directory with the other generated files.
```

>    For bash scripts, the public device certificate is called ``new-device.cert.pem`` and are located in a directory called ``certs`` within the script directory.
>    The private key will be called ``new-device.key.pem`` and be located in a directory called ``private`` within the script directory.

Reviewer: ```suggestion
   For bash scripts, the public device certificate is called :file:`new-device.cert.pem` and is located in a directory called :file:`certs` within the :file:`script` directory.
   The private key is called :file:`new-device.key.pem` and located in a directory called :file:`private` within the :file:`script` directory.
```

>    Note that these names may change if Azure changes their scripts.

Reviewer: ```suggestion
   The file and directory names may change if Azure changes their scripts.
```
